crates/iota-sdk-types/src/type_tag/mod.rs

```diff
@@ -402,15 +442,68 @@ pub struct StructTag {
 }
 
 impl StructTag {
-    pub fn coin(type_tag: impl Into<TypeTag>) -> Self {
+    pub fn new_iota() -> Self {
         Self {
             address: Address::FRAMEWORK,
-            module: IdentifierRef::const_new("coin").into(),
-            name: IdentifierRef::const_new("Coin").into(),
-            type_params: vec![type_tag.into()],
+            module: IdentifierRef::const_new("iota").into(),
+            name: IdentifierRef::const_new("IOTA").into(),
+            type_params: vec![],
         }
     }
 
+    pub fn new_gas_coin() -> Self {
+        Self::new_coin(Self::new_iota())
+    }
+
+    pub fn new_id() -> Self {
+        Self {
+            address: Address::FRAMEWORK,
+            module: IdentifierRef::const_new("object").into(),
+            name: IdentifierRef::const_new("ID").into(),
+            type_params: vec![],
+        }
+    }
+
+    pub fn new_uid() -> Self {
+        Self {
+            address: Address::FRAMEWORK,
+            module: IdentifierRef::const_new("object").into(),
+            name: IdentifierRef::const_new("UID").into(),
+            type_params: vec![],
+        }
+    }
+
+    pub fn new_name(address: Address) -> Self {
+        Self {
+            address,
+            module: IdentifierRef::const_new("name").into(),
+            name: IdentifierRef::const_new("Name").into(),
+            type_params: vec![],
+        }
+    }
+
+    add_struct_tag_ctor!(FRAMEWORK, "coin", "Coin", "from_type_tag");
```

Comment:
Matter of taste but I would prefer 3 macros, a string argument to differentiate them is a bit weird IMO

Reply:
Plus it would be consistent with the FFI where you are using 3 different ones instead of this